lib/room.rb

```diff
@@ -0,0 +1,38 @@
+require_relative 'reservation'
+
+module Hotel_System
+  class Room
+    attr_reader :room_number, :reservations, :rate_adjustor
+    def initialize(room_number, rate_adjustor = 1)
+      raise ArgumentError.new("Rate_adjustor must be numberic.") unless (rate_adjustor * 1.0).is_a? Float
+      @room_number = room_number
+      @reservations = []
+      @rate_adjustor = rate_adjustor
+    end
+
+    def available?(check_in, check_out)
+      overlap = []
+      (check_in...check_out).each do |date|
+        overlap += self.reservations.select do |reservation|
+          (reservation.check_in...reservation.check_out).cover?(date)
```

Comment:
Not a big deal at all because I think it's fairly contained in this project, BUT, there could be some refactoring with this line.

Hotel is reaching into each reservation and performing the logic on if the reservation is in this date. It might be interesting to pull this into a small helper function so that reservation does it itself, so that in hotel it could be:
```
reservation.covers?(date)
```
and in `Reservation` there's a method
```
def covers? date
  (@check_in...@check_out).cover?(date)
end
```